src/appengine/libs/issue_management/jira/__init__.py

```diff
@@ -0,0 +1,207 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Jira issue tracker."""
+
+from libs.issue_management import issue_tracker
+from libs.issue_management.jira.issue_tracker_manager import (
+    IssueTrackerManager)
+from config import db_config
+
+import datetime
+
+
+class Issue(issue_tracker.Issue):
+  """Represents an issue."""
+
+  def __init__(self, itm, jira_issue):
+    self.itm = itm
+    self.jira_issue = jira_issue
+
+    self._ccs = issue_tracker.LabelStore(self.itm.get_watchers(self.jira_issue))
+    self._components = issue_tracker.LabelStore(
+        self.jira_issue.fields.components)
+    self._labels = issue_tracker.LabelStore(self.jira_issue.fields.labels)
+
+  @property
+  def issue_tracker(self):
+    """The IssueTracker for this issue."""
+    return IssueTracker(self.itm)
+
+  @property
+  def id(self):
+    """The issue identifier."""
+    return self.jira_issue.key
+
+  @property
+  def title(self):
+    """The issue title."""
+    return self.jira_issue.fields.summary
+
+  @title.setter
+  def title(self, new_title):
+    self.jira_issue.fields.summary = new_title
+
+  @property
+  def reporter(self):
+    """The issue reporter."""
+    return self.jira_issue.fields.reporter
+
+  @reporter.setter
+  def reporter(self, new_reporter):
+    self.jira_issue.fields.reporter = new_reporter
+
+  @property
+  def is_open(self):
+    """Whether the issue is open."""
+    if self.jira_issue.resolution not in ['Closed', 'Done', 'Resolved']:
+      return True
+    return False
+
+  @property
+  def closed_time(self):
+    res_date = self.jira_issue.fields.resolutiondate
+    if res_date is not None:
+      date = res_date[:res_date.index('T')]
+      return datetime.datetime.strptime(date, '%Y-%m-%d')
```

Comment:
I'm assuming the date is the most granular we can get here?

Reply:
No, actually - I was looking at the Monorail tests and didn't test my assumption that you needed it to be y-m-d elsewhere in the code, heh. I'll return the hours/minutes/seconds too.

Reply:
Oh great, UTC conversions.